### ISSUE TITLE
E2E testing strategy

### ISSUE THREAD

**Issue by mpetrun5** [ChainSafe/chainbridge-core#120]
## Implementation details

We are currently planning running integration tests when code hits main 
and run them via trigger to chainbridge-core-example repository.

There are a couple of downsides to this approach:
 - we are planning for chainbridge-core to be modular and easily extendable and this way we make it more difficult for 
module developers (e.g chainbridge-celo-module) to run a testing suite specific to that module and easily integrate into
the rest of chainbridge-core modules
 - modules for different versions of chainbridge-core will be a lot harder to test
 - it is harder for external contributors to make PRs as we don't have an easy way to test their changes when raising PRs
instead of running e2e test suite directly on every PR on chainbridge-core for directly supported modules of chainbridge-core
 - it is hard to know when some test changes that are going to be merged are valid and will pass when coming into main branch
from external and internal contributors
 - when we make some breaking changes or we add handlers for testing to core we will first need to merge into main where pipeline will fail and then create PR into chainbridge-core-example to add or change the funcionality and then rerun the pipeline which is quite an annoying process
 - we will need to make a lot of PRs that just fix tests

## Testing details

Proposed solution: 
 - we would create testing suites for all supported networks (evm2evm, evm2substrate) and test utils on chainbridge-core (like we have now) to run them easily from external modules (e.g someone external needs chainbridge-networkB-module and they can have integration test suite setup with just a few lines and security that the module is compatible with chainbridge-core ) and implement that on chainbridge-celo-module as example
 - we would have tests for chainbridge-core supported modules ran by github-actions on every PR (probably have modules 
 tested by separate workflows in parallel for easy readability of what failed and what is successful and to be a bit faster)
 - have final happy case integration test suite run on chainbridge-core-example (some roundtrip probably) of all internally supported modules

## Acceptance Criteria

All tests pass

@P1sar @MakMuftic @waymobetta @tcar121293 @nmlinaric  
Would love to hear if this makes any sense.
